### PR TITLE
feat(atomic): add result templates to instant results

### PR DESCRIPTION
**Pull request by liviarett**
https://coveord.atlassian.net/browse/KIT-1647

### CODE REVIEW

**Review by olamothe — APPROVED** [review]

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/search-box-suggestions/atomic-search-box-instant-results/atomic-search-box-instant-results.tsx [hunk 1]
```diff
@@ -23,9 +29,24 @@ export class AtomicSearchBoxInstantResults {
   @Element() private host!: HTMLElement;
 
   @State() public error!: Error;
+  @State() public templateHasError = false;
   private instantResults!: InstantResults;
 
   private results: Result[] = [];
+  public resultListCommon!: ResultListCommon;
```
[the remark this answers is not in the document]
Comment (liviarett) [reply packages/atomic/src/components/search-box-suggestions/atomic-search-box-instant-results/atomic-search-box-instant-results.tsx:36]: for this component, I think so!

File: packages/atomic/src/pages/examples/suggestions.html [hunk 2]
```diff
@@ -83,7 +83,13 @@
           <atomic-search-box suggestion-timeout="1000">
             <atomic-did-you-mean></atomic-did-you-mean>
             <atomic-search-box-recent-queries icon="assets://xml.svg"> </atomic-search-box-recent-queries>
-            <atomic-search-box-instant-results></atomic-search-box-instant-results>
+            <atomic-search-box-instant-results>
```
[the remark this answers is not in the document]
Comment (liviarett) [reply packages/atomic/src/pages/examples/suggestions.html:86]: yes, it does work but I had turned it off with `includeDefaultTemplate: false`. my reasoning was that we might want to exclude certain types of results - if we allow for the default templates, anything that doesn't match your custom templates will render with the default.
I'll turn it back on in the next commit and you let me know if this comment ^ changed your mind and I should turn it off again :)

File: packages/atomic/src/components/search-box-suggestions/atomic-search-box-instant-results/atomic-search-box-instant-results.tsx [hunk 3]
```diff
@@ -23,9 +29,24 @@ export class AtomicSearchBoxInstantResults {
...
+  /**
```
[the remark this answers is not in the document]
Comment (liviarett) [reply packages/atomic/src/components/search-box-suggestions/atomic-search-box-instant-results/atomic-search-box-instant-results.tsx:37]: ok, this part still confuses me, but I'll give it a try!!

[the remark this answers is not in the document]
Comment (liviarett) [reply packages/atomic/src/components/search-box-suggestions/atomic-search-box-instant-results/atomic-search-box-instant-results.tsx:37]: done in a separate PR

File: packages/atomic/src/pages/examples/suggestions.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (olamothe) [reply packages/atomic/src/pages/examples/suggestions.html:86]: I think it should be turned on yes. Reasoning is that it's better as a dev to see something by default, and then refine afterwards to filter our content/make it look better, than to not see anything by default.

**Pull request merged**